dspace-api/src/main/java/org/dspace/app/util/Util.java

```diff
@@ -489,66 +489,57 @@ public static String getSourceVersion()
         // Read the input form file for the specific collection
         DCInputsReader inputsReader = new DCInputsReader(formFileName);
 
-        DCInputSet inputSet = inputsReader.getInputs(col_handle);
+        List<DCInputSet> inputSets = inputsReader.getInputsByCollectionHandle(col_handle);
 
-        // Replace the values of Metadatum[] with the correct ones in case of
-        // controlled vocabularies
-        String currentField = schema + "." + element
-                + (qualifier == null ? "" : "." + qualifier);
+		for (DCInputSet inputSet : inputSets) {
+			// Replace the values of Metadatum[] with the correct ones in case
+			// of
+			// controlled vocabularies
+			String currentField = schema + "." + element + (qualifier == null ? "" : "." + qualifier);
 
-        if (inputSet != null)
-        {
+			if (inputSet != null) {
 
-            int pageNums = inputSet.getNumberPages();
+				int fieldsNums = inputSet.getNumberFields();
 
-            for (int p = 0; p < pageNums; p++)
-            {
+				for (int p = 0; p < fieldsNums; p++) {
 
-                DCInput[] inputs = inputSet.getPageRows(p, false, false);
+					DCInput[] inputs = inputSet.getFields();
 
-                if (inputs != null)
-                {
+					if (inputs != null) {
 
-                    for (int i = 0; i < inputs.length; i++)
-                    {
-                        String inputField = inputs[i].getSchema()
-                                + "."
-                                + inputs[i].getElement()
-                                + (inputs[i].getQualifier() == null ? "" : "."
-                                        + inputs[i].getQualifier());
-                        if (currentField.equals(inputField))
-                        {
-
-                            myInputs = inputs[i];
-                            myInputsFound = true;
-                            break;
-
-                        }
-                    }
-                }
-                if (myInputsFound)
-                    break;
-            }
-        }
+						for (int i = 0; i < inputs.length; i++) {
+							String inputField = inputs[i].getSchema() + "." + inputs[i].getElement()
+									+ (inputs[i].getQualifier() == null ? "" : "." + inputs[i].getQualifier());
```

Comment:
Yes, I agree. So I will introduce the new method DCInput#getFieldName that build the description of the field based on schema.element.qualifier. To add into the Change Log for the migration because will be used from a public method (https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/app/util/SubmissionConfigReader.java#L109 and https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/app/util/DCInputSet.java#L126).

Reply:
👍

Reply:
Do you still need change this?